fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java

```diff
@@ -937,7 +941,33 @@ protected JobPayloadBuilder buildJobPayloadBuilder() {
      * @return A DruidWebService
      */
     protected DruidWebService buildDruidWebService(DruidServiceConfig druidServiceConfig, ObjectMapper mapper) {
-        return new AsyncDruidWebServiceImpl(druidServiceConfig, mapper);
+        Supplier<Map<String, String>> supplier = buildDruidWebServiceHeaderSupplier();
+        return new AsyncDruidWebServiceImpl(druidServiceConfig, mapper, supplier);
+    }
+
+    /**
+     * Build the Supplier for Druid data request headers.
+     *
+     * @return The Druid data request header Supplier.
+     */
+    protected Supplier<Map<String, String>> buildDruidWebServiceHeaderSupplier() {
+        Supplier<Map<String, String>> supplier = HashMap::new;
+        String customSupplierClassString = SYSTEM_CONFIG.getStringProperty(DRUID_HEADER_SUPPLIER_CLASS, null);
+        if (customSupplierClassString != null && customSupplierClassString != "") {
+            try {
+                Class<?> c = Class.forName(customSupplierClassString);
+                Constructor<?> constructor = c.getConstructor();
+                supplier = (Supplier<Map<String, String>>) constructor.newInstance();
+            } catch (Exception e) {
+                LOG.error(
```

Comment:
We _were_ matching something more specific in an earlier version of this PR and I suggested replacing that list of roughly 5 exceptions with just `Exception`. Since we're re-throwing the exception anyways, catching everything so that we can log it (since it'll be fatal anyways) is (I think) fine, and lets the code be smaller / cleaner.